### PR TITLE
Refactor User selection in group editing

### LINKED ISSUE #725: User selection in group editing has an api call for each selection - SP 0.5
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Each time a user is selected or deselected using the checkbox in the dialog, an api call for group.addUser or group.removeUser is bein sent. 

## Expected Behavior 🤔

Users to be added and removed should be stored (e.g. in an array), and the requests should only be sent when clicking the "Done" button.

## Steps to Reproduce 🕹

Steps:

1. open the edit dialog of a group
2. click on the dropdown for selecting users
3. try to select or deselect users and look at what calls are being sent

## Your Environment 🌎

| Tech           | Version |
| -------------- | ------- |
| TruBudget       | v1.17.0  |

### PR DESCRIPTION
**Pull request by daniel-arnauer**
Currently, only one user is can be sent to add or remove a user from a group. For better performance, send an array of user ids to the API

- [x] Refactor Postman collection json
- [x] Refactor Unit tests in API
- [x] Refactor UI
- [x] Refactor provisioning

Closes #725

### CODE REVIEW

**Comment by Stezido** [comment]
Closing PR - other focus (RoadMap)

File: api/src/index.ts [hunk 1]
```diff
@@ -393,13 +393,13 @@ GroupListAPI.addHttpHandler(server, URL_PREFIX, {
...
 
 GroupMemberAddAPI.addHttpHandler(server, URL_PREFIX, {
-  addGroupMember: (ctx, issuer, groupId, newMember) =>
-    GroupMemberAddService.addMember(db, ctx, issuer, groupId, newMember),
+  addGroupMembers: (ctx, issuer, groupId, newMembers) =>
+    GroupMemberAddService.addMembers(db, ctx, issuer, groupId, newMembers),
 });
 
 GroupMemberRemoveAPI.addHttpHandler(server, URL_PREFIX, {
-  removeGroupMember: (ctx, issuer, groupId, newMember) =>
```
Comment (mayrmartin) [inline api/src/index.ts:401]: good catch 😉

File: api/src/service/domain/organization/group_eventsourcing.ts [hunk 2]
```diff
@@ -85,79 +85,89 @@ function handleGroupCreated(
...
+    }
+    newMembers.push(member);
+  });
+
+  if (newMembers.length === 0) {
     return;
   }
 
-  group.members.push(memberAdded.newMember);
+  group.members.push(...newMembers);
```
Comment (mayrmartin) [inline api/src/service/domain/organization/group_eventsourcing.ts:111]: use array.reduce

File: api/src/service/domain/organization/group_eventsourcing.ts [hunk 3]
```diff
@@ -85,79 +85,89 @@ function handleGroupCreated(
...
+  membersRemoved.members.forEach((member) => {
+    const memberIdx = group.members.indexOf(member);
+    if (memberIdx === -1) {
+      // The "member" already doesn't belong to this group, so there's nothing left to do.
+      logger.trace("The member to remove does not belong to this group => all good!");
+      return;
+    }
+    // Remove the user from the array:
+    group.members.splice(memberIdx, 1);
+  });
```
Comment (mayrmartin) [inline api/src/service/domain/organization/group_eventsourcing.ts:151]: consider using via map & filter

```

export function remove(...itemsToRemoe) {
    return this.filter(item => !itemsToRemoe.includes(item))
}

const toRemove = membersRemoved.members.map(member => group.members.indexOf(member)).filter(idx => idx !==  - 1);

const result = group.members::remove(toRemove)
```

**Review by mayrmartin — CHANGES_REQUESTED** [review]
since endpoints changed pls update load-test endpoints (can be generated via API specification)

File: api/src/service/domain/organization/group_eventsourcing.ts (hunk 2 above)
Comment (daniel-arnauer) [reply api/src/service/domain/organization/group_eventsourcing.ts]: done

File: api/src/service/domain/organization/group_eventsourcing.ts (hunk 3 above)
Comment (daniel-arnauer) [reply api/src/service/domain/organization/group_eventsourcing.ts]: done

**Review by mayrmartin — APPROVED** [review]

**Pull request merged**